### ISSUE TITLE
[Bug] Stringified subclass forward references (e.g., `"type[MuhClass]"`) unsupported

### ISSUE THREAD

**Issue by patrick-kidger** [beartype/beartype#289]
```python
import beartype

@beartype.beartype
def f(x: "type[B]"):
    pass

class B:
    pass

f(B)
```
produces
```python
beartype.roar.BeartypeCallHintParamViolation: Function __main__.f() parameter x=<class '__main__.B'> violates type hint type[beartype._check.forward._fwdref._BeartypeForwardRefIndexable], as type <class '__main__.B'> not subclass of <class "beartype._check.forward._fwdref._BeartypeForwardRefIndexable">.
```
Python 3.11; beartype 0.16.2.

Enjoy!

**Comment by leycec** [comment]
**Yay!** It's everyone's favourite Google X researcher, @patrick-kidger. :hugs: 

> Enjoy!

You say that so glibly. Yet I am crying instead. :sob: 

Unsurprisingly, you're right. @beartype should *absolutely* support this. I'll patch this up lickety-split for PEP 563 bros everywhere. If I succeed against the overwhelming odds, I'll also push out a new @beartype 0.16.3 release early next week.

As always, thanks for the endless litany of thoughtful issues. You're forcing @beartype to improve despite its preference to just play video games, ride bicycles, and play with cats in this eerily warm Autumn air. What is this: Canada the Bahamas?

<sup>*I don't even know anymore.*</sup>

**Comment by leycec** [comment]
**Resolved by ad0757c.** Have an awesome Thanksgiving weekend – wherever this weekend may find you and your family. Thanks again for all the tasty and very instructive issues, Kidger-sama. This turkey leg's for you. :turkey:

**Comment by patrick-kidger** [comment]
Haha! Thank you for the quick fix. :)
